### ISSUE TITLE
Bug: access to file in OneDrive with Office apps installed and logged in

### ISSUE THREAD

**Issue by l-c-g** [PhilippC/keepass2android#153]
I recently moved a kdbx file to OneDrive and tried to open it with Keepass2Android. It worked fine on my phone, but did not work on my tablet. 

After I select OneDrive, it goes to a white activity with a spinning animation, then drops me back to the main activity where I can open or create a database. A brief popup at the bottom tells me "authentication not successful". I realized that I had a number of MS Office apps installed and also had an "Office" account on the tablet that the phone did not have. When I removed that account, I could login from KP2A and open the file just fine. Afterwards, I had another "Office" account on the tablet, but now I can no longer login to my account from within the office apps (I tried Word and Excel).

**Comment by l-c-g** [comment]
## Detailed steps to reproduce ##

* Install the Word app
* Login to your Microsoft Account from within Word
* Install KP2A (the cloud version)
* Open KP2A and choose to open a database
* Select OneDrive from the list of sources
* Observe the error
* In your device's settings, go to the 'Accounts' section and remove the 'Office'
* Retry the OneDrive source in KP2A
* Open Word again and note that you are no longer logged in
* Try to log in again and observe the error

**Comment by PhilippC** [comment]
this should be fixed in the latest beta:
Please switch to beta and install 1.07b-pre2 (just published to beta, see https://play.google.com/apps/testing/keepass2android.keepass2android). Please click "Change database" (on the password screen), then "Open database", "One Drive", "All files and shared files" (or something else if appropriate) and browse to your file again. This will then use the new implementation. (If you don't re-open the database like this, the old implementation will be used).
Please re-open (or comment) if the issue persists for some reason.

**Comment by l-c-g** [comment]
It kind of worked with 1.07b-pre2 but it wasn't quite smooth and I wanted to give you all the details I can, to make sure it works as intended and there are no lurking problems.

* I downloaded and installed the beta (clean on this phone, since I actually stopped using Keepass a while ago).
* The phone has both Excel and OneDrive installed and the phone has my Microsoft account in Android's sytem setting for both.
* I clicked "Open database" and "OneDrive" in KP2A and I got a normal Microsoft login, most likely OAuth.
  * This is where it gets weird: After loggin in with user and password and granting KP2A access, I get a blank page with only a single line of text "Too many requests".
* I tried again and this time I notice that it has discovered my account on the phone and offers to use this one (most likely the first time as well, I just didn't see it); I go ahead and have to grant KP2A access again.
  * This time, it works and I can browse my OneDrive files.
